### PR TITLE
[2/n]: Replace custom hash map with std::unordered_map in recv_sys

### PR DESCRIPTION
**Pull request by RahulKushwaha**

### CODE REVIEW

File: log/log0recv.cc [hunk 1]
```diff
@@ -157,7 +157,7 @@ void recv_sys_close(void) {
 void recv_sys_mem_free() {
   if (recv_sys != nullptr) {
     if (recv_sys->addr_hash != nullptr) {
```
Comment (sunbains) [inline log/log0recv.cc:159]: AFAIK, delete can take a nullptr.

File: include/log0recv.h [hunk 2]
```diff
@@ -205,8 +205,18 @@ struct recv_addr_struct {
   hash_node_t addr_hash;
 };
 
+struct Space_id_page_no_hash {
+  std::size_t operator()(const std::pair<space_id_t, page_no_t> &pair) const {
```
Comment (sunbains) [inline include/log0recv.h:209]: Make this into a new type, it's used everywhere. We can make the other changes later.

struct Page_id {
  space_id_t m_space;
  page_no_t m_page_no;
};

Overload the <, ==, > and != operator too and add is_null() and to_string() method to it.

File: include/buf0types.h [hunk 3]
```diff
@@ -521,50 +521,6 @@ struct buf_pool_stat_t {
   ulint n_pages_not_made_young{};
 };
 
-struct Page_id {
```
Comment (RahulKushwaha) [inline include/buf0types.h:524]: I guess you already created `Page_id`. Moved it to `innodb0types`.

File: include/buf0types.h [hunk 4]
```diff
@@ -521,50 +521,6 @@ struct buf_pool_stat_t {
...
-  }
-
-  /** @return true if values have not been set. */
-  bool is_null() const {
-    if (m_space_id == NULL_SPACE_ID) {
-      /* Both must be null or not null. */
-      ut_a(m_page_no == NULL_PAGE_NO);
-      return true;
-    } else {
-      ut_a(m_page_no != NULL_PAGE_NO);
```
Comment (RahulKushwaha) [inline include/buf0types.h:554]: @sunbains : Remove these assertions in `innodb0types` as `ut` was not available. Should we add the header there if we want to keep these?

**Comment by sunbains** [comment]
I fixed this, along with some other changes.

**Pull request closed without merging**